DuckDuckGo/TabViewController.swift

```diff
@@ -1046,18 +1036,18 @@ extension TabViewController: WKNavigationDelegate {
         }
 
         if navigationAction.navigationType == .linkActivated, let url = navigationAction.request.url {
-            switch tapLinkDestination {
-            case .newTab:
-                decisionHandler(.cancel)
-                delegate?.tab(self, didRequestNewTabForUrl: url, openedByPage: false)
-                return
-
-            case .backgroundTab:
-                decisionHandler(.cancel)
-                delegate?.tab(self, didRequestNewBackgroundTabForUrl: url)
-                return
-                
-            default: break
+            let modifierFlags = delegate?.tabWillRequestNewTab(self)
```

Comment:
I'd put this directly to the 'if' above (line 1038), so we can have 'if let' construct and avoid working with optionals.